pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -159,33 +162,15 @@ func (c *Cluster) Start() error {
 		}
 	}
 
-	if err := c.configureDashboardService(); err != nil {
-		logger.Errorf("failed to enable dashboard. %v", err)
+	if err := c.reconcileServices("", daemonIDs[0]); err != nil {
```

Comment:
So we default to the first in the set, do we have any guarantee that it will be the active manager? Or we just don't care the `reconcileServices()` will take care of that? Also, should we wait a bit for the deployments to be ready? I'm thinking that we might end up in this condition:

```go
	activeDaemon := mgrMap.ActiveName
	if activeDaemon == "" {
		if defaultMgr == "" {
			return errors.Errorf("active mgr not found")
```

If the daemons have not started yet, especially on a multi-node setup if the mgr pod ends up on the machine which does not have the image.

Reply:
> So we default to the first in the set, do we have any guarantee that it will be the active manager? Or we just don't care the reconcileServices() will take care of that? 

The services will be created basically as a placeholder with mon-a as the active mgr. If it doesn't end up being the active one, you're right that the sidecar will take care of it very soon.

> Also, should we wait a bit for the deployments to be ready?
> If the daemons have not started yet, especially on a multi-node setup if the mgr pod ends up on the machine which does not have the image.

If the sidecar doesn't detect an active mgr it just log a message and will try again in 15s. When ceph finally responds with the active mgr, then the sidecar will update it.

Reply:
Ah yes I see it now, the caller logs the error and retries.